nixos/modules/system/boot/loader/generic-extlinux-compatible/default.nix

```diff
@@ -45,6 +45,20 @@ in
         '';
       };
 
+      showPrompt = mkOption {
+        default = true;
+        type = types.bool;
+        description = ''
+          Whether to prompt for user selection in boot menu.
+
+          When enabled, the bootloader will show a menu allowing to select
+          a generation to boot.
+
+          Disabling the prompt will not show the menu and always boot into
+          the default boot entry regardless of timeout or configuration limit.
+        '';
+      };
+
```

Comment:
Hijacking this line for a sub-thread

* * *

The way the different "extlinux-compatible" bootloadery bits behave is a total mess.

* * *

From syslinux `txt/syslinux.cfg.txt`


```
*PROMPT* 'flag_val'::                                                 
If 'flag_val' is 0, display the boot: prompt only if the Shift or Alt
key is pressed, or Caps Lock or Scroll lock is set (this is the   
default).  If 'flag_val' is 1, always display the boot: prompt.
```

I haven't looked at the source to confirm, but it should be assumed its default really is `0`.

And at `0` the prompt will only be shown if modifiers are held/set.

* * *

From U-Boot `common/menu.c`

```
timeout - A delay in seconds to wait for user input. If 0, timeout is
disabled, and the default choice will be returned unless prompt is 1.

prompt - If 0, don't ask for user input unless there is an interrupted    
timeout. If 1, the user will be prompted for input regardless of the value
of timeout.                                                               
```

~~Side-note, in U-Boot, the default is `0` unless `MENU` is present in the config, which will set `prompt = 1`. So in practice since we have `MENU` it should set `prompt = 1`.~~ Uh, I need to review this statement, since there's the thing about interrupted timeout... :melting_face: 

Though this all should be taken with a big pile of salt, since [the commit referenced](https://source.denx.de/u-boot/u-boot/-/commit/739e8361f3fe78038251216df6096a32bc2d5839) implies the documented behaviour might not have been matching reality. It is unclear from the commit alone if this was from a regression, or it never worked as documented.

* * *

# :melting_face: 

Considering both of those differ in behaviour, *and also* historically U-Boot may have differed in behaviour, I'm not sure adding outright `PROMPT` is a good idea.

We might want to make it a nullable option. Leave its default to `null`, and only add `PROMPT` when a conscious choice is made.

@dasJ thoughts?

Reply:
I'm okay with implementing this as a three-state thing and leave the default `null`. If @dasJ has no objections, I will update this PR.